src/main/java/org/deepsymmetry/beatlink/MediaDetails.java

```diff
@@ -122,6 +120,34 @@ private int getUTF16StringLength(byte[] source, int offset, int maxLength) {
         return numBytes;
     }
 
+    /**
+     * Constructor sets all the immutable interpreted fields based on constructor inputs
+     *
+     * @param packet the media response packet that was received
+     */
+
+    /**
+     * Custom constructor to emulate an actual MediaDetails object from CDJs. This allows our Status packets
+     * to use OpusProvider to enrich the track data.
+     *
+     * @param slotReference Slot Reference to Emulate
+     * @param mediaType Media Type to Emulate
+     * @param name Name of device
+     */
+    public MediaDetails(SlotReference slotReference, CdjStatus.TrackType mediaType, String name) {
+        this.slotReference = slotReference;
+        this.mediaType = mediaType;
+        this.name = name;
+        this.creationDate = "";
+        this.trackCount = 0;
```

Comment:
We should probably supply the actual counts of tracks and playlists available from the mounted metadata archive.

Reply:
Oh, using the OpusProvider you mean? This makes sense, setting it up.

Reply:
Ok, I have filled in the stuff I could find with reasonable details. One thing I cannot find in totalSize and freeSpace. I just put 0.

Reply:
Yeah, you put in more than I expected, this is great! 😄